packages/notebook-extension/src/index.ts

```diff
@@ -2243,9 +2243,7 @@ function addCommands(
   });
   commands.addCommand(CommandIDs.duplicateBelow, {
     label: trans.__('Duplicate Cells Below'),
-    caption: trans.__(
-      'Copy the selected cells and paste them below the selection'
-    ),
+    caption: trans.__('Duplicate cell below'),
```

Comment:
We should be smarter because they may be more than one cell selected.

Could you add a new attribute `selectedCells` to `Notebook` in https://github.com/jupyterlab/jupyterlab/blob/1250d94d3dc91213e6ea18e4005705d88f17aa1a/packages/notebook/src/widget.ts#L1206

It should return an array of active or selected cells. 

You must not compute the result every time. In `Notebook` listen for `selectionChanged` and `activeCellChanged` to update a cached list.

Then it will be possible here to change the `label` and `caption` of the commands to:

```ts
label: args => {
  const current = getCurrent(tracker, shell, args);
  return trans._n('<singular text>', '<plural text>', current?.selectedCells.length ?? 0);
}
```

Reply:
Thanks @fcollonval Commit 9567df2 propose a tentative implementation that propose a label and a caption with the correct form (singular or plural) depending on the number of active or selected cells. I observed some case where the list of selectedCell is not updated. For instance :
- when starting from one active cell and going to 2 selected cells: only the active cell is in the list
-  when going from n selected cell to n-1, the nth cell remains in the list

Reply:
The issue mentionned just above has been solved using `commands.notifyCommandChanged` in commit [55b3dfb](https://github.com/jupyterlab/jupyterlab/pull/13861/commits/55b3dfbbcfd04ff70fc9d2d56d24a107d97934a3)

```
  tracker.selectionChanged.connect(() => {
    commands.notifyCommandChanged(CommandIDs.duplicateBelow);
  });

```